test/lib/factories/PathStartFactory.test.ts

```diff
@@ -0,0 +1,194 @@
+import path from 'node:path';
+import {PathStartFactory} from '../../../src/lib/factories/PathStartFactory';
+
+
+describe('PathStartFactory implementations', () => {
+	const ORIGINAL_TEST_DIR = process.cwd();
+	const PATH_TO_WORKSPACE = path.join(__dirname, '..', '..', 'fixtures', 'example-workspaces', 'workspace-with-dotted-items');
+
+	beforeEach(() => {
+		process.chdir(PATH_TO_WORKSPACE);
+	});
+
+	afterEach(() => {
+		process.chdir(ORIGINAL_TEST_DIR);
+	})
+
+	it('When passed undefined, returns undefined', async () => {
+		const factory = new PathStartFactory();
+		const output = await factory.create(undefined);
+		expect(output).toBeUndefined();
+	});
+
+	it('When passed an empty array, returns an empty array', async () => {
+		const factory = new PathStartFactory();
+		const output = await factory.create([]);
+		expect(output).toEqual([]);
+	});
+
+	describe('File matching', () => {
+		it.each([
+			{scenario: 'a single relative file', expectation: 'just that file', input: ['undotted-file-1.txt']},
+			{scenario: 'a list of relative files', expectation: 'just those files', input: [path.join('.', 'undotted-directory-a', 'undotted-file-2a.txt'), '.dotted-file-2.txt']},
+			{
+				scenario: 'a single absolute file',
+				expectation: 'just that file',
+				input: [path.join(PATH_TO_WORKSPACE, 'undotted-file-1.txt')]
+			},
+			{
+				scenario: 'a list of absolute files',
+				expectation: 'just those files',
+				input: [
+					path.join(PATH_TO_WORKSPACE, 'undotted-directory-a', 'undotted-file-2a.txt'),
+					path.join(PATH_TO_WORKSPACE, 'undotted-file-2.txt')
+				]
+			}
+		])('Given $scenario, returns $expectation', async ({input}) => {
+			const factory = new PathStartFactory();
+			const output = await factory.create(input);
+			expect(output).toBeDefined();
+			expect(output as string[]).toHaveLength(input.length);
+			expect(output as string[]).toEqual(input);
+		})
+	});
+
+	describe('Glob matching', () => {
+		it.each([
+			{
+				scenario: 'a single relative file',
+				glob: path.join('.', 'undotted*1.txt'),
```

Comment:
NOTE: Technically a glob should always be UNIX formatted even on Windows, but due to our use of `normalize-path`, we actually support standard Globs and Windows-formatted pseudo-globs.